File: backend/internal/manga/manga.go
package manga

import (
	"time"
)

type Status string

const (
	StatusReading   Status = "Reading"
	StatusBacklog   Status = "Backlog"
	StatusCompleted Status = "Completed"
	StatusDropped   Status = "Dropped"
	StatusHiatus    Status = "Hiatus"
)

type Manga struct {
	ID        string    `json:"id"`
	Title     string    `json:"title"`
	Image     *string   `json:"image"`
	URL       *string   `json:"url"`
	Status    Status    `json:"status"`
	Chapter   int       `json:"chapter"`
	AddedAt   time.Time `json:"added_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

func (m *Manga) Validate() error {
	if m.Title == "" {
		return ErrInvalidTitle
	}
	if m.Chapter < 0 {
		return ErrInvalidChapter
	}
	switch m.Status {
	case StatusReading, StatusBacklog, StatusCompleted, StatusDropped, StatusHiatus:
		// valid
	default:
		return ErrInvalidStatus
	}
	return nil
}

File: frontend/src/types/manga.ts
export type MangaStatus = 'Reading' | 'Backlog' | 'Completed' | 'Dropped' | 'Hiatus'

export interface Manga {
  id: string
  title: string
  image: string | null
  url: string | null
  status: MangaStatus
  chapter: number
  added_at: string
  updated_at: string
}

export interface CreateMangaRequest {
  title: string
  image?: File
  url?: string
  status: MangaStatus
  chapter: number
}

export interface UpdateMangaRequest {
  title?: string
  image?: File
  url?: string
  status?: MangaStatus
  chapter?: number
}
